src/action/macos/create_fstab_entry.rs

```diff
@@ -236,76 +148,67 @@ impl Action for CreateFstabEntry {
+            .map_err(|e| Self::error(ActionErrorKind::Open(fstab_path.to_owned(), e)))?;
 
-            let mut file = OpenOptions::new()
-                .create(false)
-                .write(true)
-                .read(true)
-                .open(&fstab_path)
-                .await
-                .map_err(|e| Self::error(ActionErrorKind::Open(fstab_path.to_owned(), e)))?;
-
-            let mut file_contents = String::default();
-            file.read_to_string(&mut file_contents)
-                .await
-                .map_err(|e| Self::error(ActionErrorKind::Read(fstab_path.to_owned(), e)))?;
+        let mut fstab_buf = String::new();
+        fstab
+            .read_to_string(&mut fstab_buf)
+            .await
+            .map_err(|e| Self::error(ActionErrorKind::Read(fstab_path.to_owned(), e)))?;
 
-            if let Some(start) = file_contents.rfind(fstab_entry.as_str()) {
-                let end = start + fstab_entry.len();
-                file_contents.replace_range(start..end, "")
-            }
+        let updated_buf = fstab_buf
+            .lines()
+            .filter_map(|line| {
+                // Delete nix-installer entries with a "prelude" comment
+                if line.starts_with("# nix-installer created volume labelled") {
+                    None
+                } else {
+                    Some(line)
+                }
+            })
+            .filter_map(|line| {
+                if line.split(&[' ', '\t']).nth(2) == Some("/nix") {
+                    // Delete the mount line for /nix
+                    None
+                } else {
+                    Some(line)
+                }
+            })
+            .collect::<Vec<&str>>()
+            .join("\n");
 
-            file.seek(SeekFrom::Start(0))
-                .await
-                .map_err(|e| Self::error(ActionErrorKind::Seek(fstab_path.to_owned(), e)))?;
-            file.set_len(0)
-                .await
-                .map_err(|e| Self::error(ActionErrorKind::Truncate(fstab_path.to_owned(), e)))?;
-            file.write_all(file_contents.as_bytes())
-                .await
-                .map_err(|e| Self::error(ActionErrorKind::Write(fstab_path.to_owned(), e)))?;
-            file.flush()
-                .await
-                .map_err(|e| Self::error(ActionErrorKind::Flush(fstab_path.to_owned(), e)))?;
-        } else {
-            return Err(Self::error(CreateFstabEntryError::CannotDetermineFstabLine));
-        }
+        fstab
+            .seek(SeekFrom::Start(0))
+            .await
+            .map_err(|e| Self::error(ActionErrorKind::Seek(fstab_path.to_owned(), e)))?;
+        fstab
+            .set_len(0)
+            .await
+            .map_err(|e| Self::error(ActionErrorKind::Truncate(fstab_path.to_owned(), e)))?;
+        fstab
+            .write_all(updated_buf.as_bytes())
+            .await
+            .map_err(|e| Self::error(ActionErrorKind::Write(fstab_path.to_owned(), e)))?;
```

Comment:
I wonder if we should replace this with our `write-tmp-then-move` approach while we're here.

Reply:
Good idea, done!